resources/packs/gcp/compute.go

```diff
@@ -1864,59 +1865,129 @@ func (g *mqlGcpProjectComputeServiceAddress) GetNetwork() (interface{}, error) {
-	parts := strings.Split(params, "/")
-	resId := resourceId{Project: parts[1], Region: parts[3], Name: parts[5]}
+	ctx := context.Background()
+	fwrSvc, err := computev1.NewForwardingRulesRESTClient(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		return nil, err
+	}
 
-	return g.MotorRuntime.CreateResource("gcp.project.computeService.subnetwork",
-		"name", resId.Name,
-		"projectId", resId.Project,
-		"region", resId.Region,
-	)
+	var fwRules []interface{}
+	it := fwrSvc.AggregatedList(ctx, &computepb.AggregatedListForwardingRulesRequest{Project: projectId, IncludeAllScopes: ptr.Bool(true)})
+	for {
+		resp, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		for _, fwr := range resp.Value.ForwardingRules {
+			metadataFilters := make([]interface{}, 0, len(fwr.GetMetadataFilters()))
+			for _, m := range fwr.GetMetadataFilters() {
+				filterLabels := make([]interface{}, 0, len(m.GetFilterLabels()))
+				for _, l := range m.GetFilterLabels() {
+					filterLabels = append(filterLabels, map[string]interface{}{
+						"name":  l.GetName(),
+						"value": l.GetValue(),
+					})
+				}
+				metadataFilters = append(metadataFilters, map[string]interface{}{
+					"filterLabels":        filterLabels,
+					"filterMatchCriteria": m.GetFilterMatchCriteria(),
+				})
+			}
+
+			serviceDirRegs := make([]interface{}, 0, len(fwr.GetServiceDirectoryRegistrations()))
+			for _, s := range fwr.GetServiceDirectoryRegistrations() {
+				serviceDirRegs = append(serviceDirRegs, map[string]interface{}{
+					"namespace":              s.GetNamespace(),
+					"service":                s.GetService(),
+					"serviceDirectoryRegion": s.GetServiceDirectoryRegion(),
+				})
+			}
+			mqlFwr, err := g.MotorRuntime.CreateResource("gcp.project.computeService.forwardingRule",
+				"id", fmt.Sprintf("%d", fwr.Id),
+				"ipAddress", fwr.GetIPAddress(),
+				"ipProtocol", fwr.GetIPProtocol(),
+				"allPorts", fwr.GetAllPorts(),
+				"allowGlobalAccess", fwr.GetAllowGlobalAccess(),
+				"backendService", fwr.GetBackendService(),
+				"created", parseTime(fwr.GetCreationTimestamp()),
+				"description", fwr.GetDescription(),
+				"ipVersion", fwr.GetIpVersion(),
+				"isMirroringCollector", fwr.GetIsMirroringCollector(),
+				"labels", core.StrMapToInterface(fwr.GetLabels()),
+				"loadBalancingScheme", fwr.GetLoadBalancingScheme(),
+				"metadataFilters", metadataFilters,
+				"name", fwr.GetName(),
+				"networkUrl", fwr.GetNetwork(),
+				"networkTier", fwr.GetNetworkTier(),
+				"noAutomateDnsZone", fwr.GetNoAutomateDnsZone(),
+				"portRange", fwr.GetPortRange(),
+				"ports", core.StrSliceToInterface(fwr.GetPorts()),
+				"regionUrl", fwr.GetRegion(),
+				"serviceDirectoryRegistrations", serviceDirRegs,
+				"serviceLabel", fwr.GetServiceLabel(),
+				"serviceName", fwr.GetServiceName(),
+				"subnetworkUrl", fwr.GetSubnetwork(),
```

Comment:
I'm not sure about this one.
The method would return a `gcp.project.computeService.subnetwork` when it finds one and not the URL. Is there something missing? Or missed I something?

Reply:
in this context `fwr` is a Google Protobuf struct. `GetSubnetwork` returns the value for the `Subnetwork` field in that struct. The field contains the URL. It's just confusing because we map their field to a field with a different name. The behaviour is correct though